src/app/[locale]/client-search-dataset.ts

```diff
@@ -7,6 +7,7 @@ interface SearchDatasets {
   query?: string;
   offset: number;
   sort: Sort;
+  fetchErrorText: string;
```

Comment:
I'm not sure whether this should be a property of `clientSearchDatasets()`. Some thoughts:

1. If the function relies on an error message that's passed in externally, the function isn't really independent anymore: the caller of the function determines the error message (which may not be the message that the function itself would use).

2. If we, at some point, want the function to throw another error, we have to add another property for that one too - which doesn't scale well.

3. A localized error message is, I think, 'presentation logic' that can be better handled by a component for error display.

Would it be an idea to have `clientSearchDatasets()` throw a specific error (e.g. its own instance of `Error`?) that the error display component can use for determining the error message that needs to be presented to the user (including localization)? Or perhaps we could use an existing package here, e.g. https://www.npmjs.com/package/http-errors ?

What do you think?

Reply:
Another thought: can we make it simpler (for the time being)? For example in...

https://github.com/colonial-heritage/dataset-browser/blob/a8100debdb4133cf97519dcea592f3f4dad3c0ed/src/app/%5Blocale%5D/dataset-list.tsx#L73

... we could do the localization. Not of the `error.message` but by only displaying a generic message - 'The datasets could not be loaded' or something. We can revisit this as soon as we need to localize the specific error messages of `clientSearchDatasets()`?

Reply:
I agree. I was also not happy with the translation there but was unsure about the best alternative.